### DISCUSSION TITLE
Installation of e3nn on Linux

### DISCUSSION THREAD

**Discussion by Haliba12** [e3nn/e3nn#470] [General]
Hello,
I have been failing to install e3nn on my HPC home directory,
I have bee following a procedure which says that Pytorch must be build from source to be compatible with my machines GPUs,
But this process always fails. I was just wondering if anyone can share the e3nn installation procedure with me.
Here is the procedure below:
Pytorch must be built from source in order to be compatible with Xena’s GPU.
Supporting packages (torch_geometric, etc.) and e3nn can be installed with pip.

Different versions of e3nn don’t seem to be backwards compatible (I get errors when loading checkpoints files).

Here’s my build notes for installing e3nn v0.5.5. Instructions were adapted from:
https://github.com/pytorch/pytorch#from-source

conda install python=3.9

Get Pytorch source:

git clone --recursive https://github.com/pytorch/pytorch
cd pytorch

(
Optional:
To clear an error, I needed to add to caffe2/utils/math_gpu.cu:

#include <thrust/host_vector.h> 
)

Install prerequisites:

Look at README.MD and follow instructions to install prerequisites. This looks something like:

conda install cmake ninja
pip install -r requirements.txt

Load CUDA compilers:

module load cuda

#build
python setup.py develop

Check that PyTorch is installed in Python:

import torch

print(torch.version.cuda)
11.4
print(torch.cuda.is_available())
True

#Install torch_geometric
# from https://pytorch-geometric.readthedocs.io/en/latest/install/installation.html

pip install --no-cache-dir torch_geometric
pip install --no-cache-dir torch_cluster
pip install --no-cache-dir torch_scatter
pip install --no-cache-dir e3nn

Extras

Install psi4 from: https://psicode.org/

Install gau2grid (used to assess machine learning error):

git clone --recursive https://github.com/dgasmith/gau2grid
cd gau2grid
python setup.py install

**Comment by mitkotak** [comment]
Hey sorry for the delay.

- Can you build directly from the main branch ?

- Can you share the error that you were getting for checkpoint files ?
